pkg/local_workflows/data_transformation_workflow.go

```diff
@@ -108,63 +108,353 @@ func dataTransformationEntryPoint(invocationCtx workflow.InvocationContext, inpu
+}
 
-	localFinding.Summary.Path = testSummary.Path
-	localFinding.Summary.Artifacts = testSummary.Artifacts
-	localFinding.Summary.Type = testSummary.Type
-	localFinding.Summary.Counts.CountKeyOrderAsc.Severity = testSummary.SeverityOrderAsc
-	localFinding.Summary.Counts.Count = 0
-	localFinding.Summary.Counts.CountAdjusted = 0
-	localFinding.Summary.Counts.CountSuppressed = 0
+func transformTestSummary(testSummary *json_schemas.TestSummary, sarifDoc *sarif.SarifDocument) *local_models.TypesFindingsSummary {
+	var summary local_models.TypesFindingsSummary
+	summary.Path = testSummary.Path
+	summary.Artifacts = testSummary.Artifacts
+	summary.Type = testSummary.Type
+	summary.Counts.CountKeyOrderAsc.Severity = testSummary.SeverityOrderAsc
+	summary.Counts.Count = 0
+	summary.Counts.CountAdjusted = 0
+	summary.Counts.CountSuppressed = 0
+	summary.Counts.CountBy.Severity = make(map[string]uint32)
+	summary.Counts.CountByAdjusted.Severity = make(map[string]uint32)
+	summary.Counts.CountBySuppressed.Severity = make(map[string]uint32)
+	summary.Counts.CountBy.AdditionalProperties = make(map[string]map[string]uint32)
+	summary.Counts.CountByAdjusted.AdditionalProperties = make(map[string]map[string]uint32)
+	summary.Counts.CountBySuppressed.AdditionalProperties = make(map[string]map[string]uint32)
 
 	for _, summaryResults := range testSummary.Results {
-		localFinding.Summary.Counts.CountBy.Severity[summaryResults.Severity] = uint32(summaryResults.Total)
-		localFinding.Summary.Counts.CountByAdjusted.Severity[summaryResults.Severity] = uint32(summaryResults.Open)
-		localFinding.Summary.Counts.CountBySuppressed.Severity[summaryResults.Severity] = uint32(summaryResults.Ignored)
+		summary.Counts.CountBy.Severity[summaryResults.Severity] = uint32(summaryResults.Total)
+		summary.Counts.CountByAdjusted.Severity[summaryResults.Severity] = uint32(summaryResults.Open)
+		summary.Counts.CountBySuppressed.Severity[summaryResults.Severity] = uint32(summaryResults.Ignored)
+
+		summary.Counts.CountAdjusted += summary.Counts.CountByAdjusted.Severity[summaryResults.Severity]
+		summary.Counts.CountSuppressed += uint32(summaryResults.Ignored)
+		summary.Counts.Count += uint32(summaryResults.Total)
+	}
 
-		localFinding.Summary.Counts.CountAdjusted += localFinding.Summary.Counts.CountByAdjusted.Severity[summaryResults.Severity]
-		localFinding.Summary.Counts.CountSuppressed += uint32(summaryResults.Ignored)
-		localFinding.Summary.Counts.Count += uint32(summaryResults.Total)
+	var coverage []local_models.TypesCoverage
+	for _, run := range sarifDoc.Runs {
+		for _, cov := range run.Properties.Coverage {
+			coverage = append(coverage, local_models.TypesCoverage{
+				Files:       cov.Files,
+				IsSupported: cov.IsSupported,
+				Lang:        cov.Lang,
+				Type:        cov.Type,
+			})
+		}
 	}
+	summary.Coverage = coverage
 
-	return localFinding, nil
+	return &summary
+}
+
+func mapLocations(res sarif.Result) []local_models.IoSnykReactiveFindingLocation {
```

Comment:
Suggestion: There are two places where we map PhysicalLocations, here and in `mapCodeFlows` following DRY, I would recommend to unite them. Either we also try to combine the types or use generics to handle different return types. But given that the field anems are the same and the input structure is the same, we should try to avoid this code duplication.